include/NAS2D/Renderer/Vector.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+namespace NAS2D {
+
+
+template <typename BaseType>
+struct Vector {
+	BaseType x = 0;
+	BaseType y = 0;
+
```

Comment:
No support for 3D or 4D vectors? How is matrix math going to be handled?

Reply:
That's not a concern at this time. The class was modelled to match existing `Point` and `Rectangle` classes. We could do this at some point, but it likely needs to involve all classes at the same time. We should defer until after the other classes have been switched over to proper structs.